Pinning sha256 in embeddedArtifactRegistry in definition file result in eib-embedded-registry.service crashing

If you pin a image to a sha256 digest in the definition file like:

```yaml
embeddedArtifactRegistry:
  images:
    - name: busybox:1.37@sha256:a5d0ce49aa801d475da48f8cb163c354ab95cab073cd3c138bd458fc8257fbf1
    - name: quay.io/nmstate/kubernetes-nmstate-handler:v0.83.0@sha256:ccb9f7575965d58133468a6fa64b6421fea5718c2009fd295904bcb6ec4aaa83
    - name: quay.io/prometheus/prometheus:v3.1.0@sha256:6559acbd5d770b15bb3c954629ce190ac3cbbdb2b7f1c30f0385c4e05104e218
    - name: quay.io/prometheus-operator/prometheus-config-reloader:v0.79.2@sha256:193280a33bc1acad9bc1956c5d986e3da6950882bda89311bac317998dcebf30
```
the `eib-embedded-registry.service` (hauler) will crash at startup:

```
Mar 27 15:20:11 hostname hauler[11430]: 2025-03-27 15:20:11 ERR Error: remote write: PUT http://127.0.0.1:44467/v2/prometheus-operator/prometheus-config-reloader/manifests/sha256:193280a33bc1acad9bc1956c5d986e3da6950882bda89311bac317998dcebf30: DIGEST_INVALID: provided digest did not match uploaded content
Mar 27 15:20:11 hostname hauler[11430]: 2025-03-27 15:20:11 ERR main.go:74: error during command execution: remote write: PUT http://127.0.0.1:44467/v2/prometheus-operator/prometheus-config-reloader/manifests/sha256:193280a33bc1acad9bc1956c5d986e3da6950882bda89311bac317998dcebf30: DIGEST_INVALID: provided digest did not match uploaded content
Mar 27 15:20:11 hostname hauler[11430]: Error: exit status 1
```

If you do a `hauler store add image` with the `--plaform` flag the sha256 digest will change, this is what EIB does when building the image. This could be a user error and sha256 digest pinning is not supported, please let me know.

--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -9,6 +9,7 @@
 * Improved Validation for `operatingSystem.enableFIPS` flag
 * Added the ability to build RAW Encrypted Images
 * Improved Embedded Artifact Registry handling to no longer be memory bound
+* Updated Embedded Artifact Registry documentation
 * Dependency upgrades
   * Go module version is now upgraded from `1.22` to `1.24`
   * Updated MetalLB from `0.14.9` to `0.1.0+up0.14.9`

--- a/docs/building-images.md
+++ b/docs/building-images.md
@@ -370,8 +370,13 @@ The following describes the possible options for the embedded artifact registry 
 embeddedArtifactRegistry:
   images:
     - name: hello-world:latest
-    - name: ghcr.io/fluxcd/flux-cli@sha256:02aa820c3a9c57d67208afcfc4bce9661658c17d15940aea369da259d2b976dd
+    - name: nginx:stable@sha256:b03c8dfc241047d827e1e14d69533205b387d476d97ef7efce58605a6c3acb84
 ```
+
+> **_NOTE:_** When providing images tagged with a `sha256` digest, the digest must be the manifest digest for the 
+> platform you are pulling the image for (e.g. `linux/amd64`), not the index digest of the image. If the index digest
+> is provided and not the manifest digest, the image build will be successful but the embedded artifact registry 
+> systemd unit will fail to start at boot time.
 
 * `images` - Defines a list of container images to download and host on the node.
   * `name` - Required; Specifies the name, with a tag or digest, of a container image to be pulled and stored.

--- a/pkg/image/definition_test.go
+++ b/pkg/image/definition_test.go
@@ -164,7 +164,7 @@ func TestParse(t *testing.T) {
 	// EmbeddedArtifactRegistry
 	embeddedArtifactRegistry := definition.EmbeddedArtifactRegistry
 	assert.Equal(t, "hello-world:latest", embeddedArtifactRegistry.ContainerImages[0].Name)
-	assert.Equal(t, "ghcr.io/fluxcd/flux-cli@sha256:02aa820c3a9c57d67208afcfc4bce9661658c17d15940aea369da259d2b976dd", embeddedArtifactRegistry.ContainerImages[1].Name)
+	assert.Equal(t, "nginx:stable@sha256:b03c8dfc241047d827e1e14d69533205b387d476d97ef7efce58605a6c3acb84", embeddedArtifactRegistry.ContainerImages[1].Name)
 
 	// Kubernetes
 	kubernetes := definition.Kubernetes

--- a/pkg/image/testdata/full-valid-example.yaml
+++ b/pkg/image/testdata/full-valid-example.yaml
@@ -81,7 +81,7 @@ operatingSystem:
 embeddedArtifactRegistry:
   images:
     - name: hello-world:latest
-    - name: ghcr.io/fluxcd/flux-cli@sha256:02aa820c3a9c57d67208afcfc4bce9661658c17d15940aea369da259d2b976dd
+    - name: nginx:stable@sha256:b03c8dfc241047d827e1e14d69533205b387d476d97ef7efce58605a6c3acb84
 kubernetes:
   version: v1.30.3+rke2r1
   network:

